database/sql/src/main/scala/cromwell/database/slick/MetadataSlickDatabase.scala

```diff
@@ -284,29 +277,30 @@ class MetadataSlickDatabase(originalDatabaseConfig: Config)
                                 buildUpdatedSummary:
                                 (Option[WorkflowMetadataSummaryEntry], Seq[MetadataEntry])
                                   => WorkflowMetadataSummaryEntry,
-                                summaryPositionFunction: Seq[MetadataEntry] => DBIO[Long],
+                                summaryPositionFunction: Seq[MetadataEntry] => Long,
                                 summaryName: String
                                )(implicit ec: ExecutionContext): DBIO[Long] = {
+
+    val exactMatchMetadataKeys = Set(
+      startMetadataKey, endMetadataKey, nameMetadataKey, statusMetadataKey, submissionMetadataKey, parentWorkflowIdKey, rootWorkflowIdKey)
+    val startsWithMetadataKeys = Set(labelMetadataKey)
+
     for {
-      metadataEntries <- dataAccess.metadataEntriesForIdRange((
+      rawMetadataEntries <- dataAccess.metadataEntriesForIdRange((
         minMetadataEntryId,
-        maxMetadataEntryId,
-        startMetadataKey,
-        endMetadataKey,
-        nameMetadataKey,
-        statusMetadataKey,
-        submissionMetadataKey,
-        parentWorkflowIdKey,
-        rootWorkflowIdKey,
-        labelMetadataKey
+        maxMetadataEntryId
       )).result
+      summaryPosition = summaryPositionFunction(rawMetadataEntries)
+      metadataEntries = rawMetadataEntries filter { entry =>
+        entry.callFullyQualifiedName.isEmpty && entry.jobIndex.isEmpty && entry.jobAttempt.isEmpty &&
```

Comment:
These extra `jobIndex` and `jobAttempt` filters seem unnecessary given that the `callFullyQualifiedName` is not set, but not actively wrong.

Reply:
I think so too but in the interest of not poking any bears unnecessarily I just copied the logic of the existing code.